controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -3721,6 +3733,120 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
       });
     },
 
+    publishPersistedOperations: (req, ctx) => {
+      /**
+       * Receives a federated graph name and a list of persisted operation contents.
+       * First, it validates that the graph exists and all the operations are valid,
+       * then it stores them. Additionally, if the provided client name for registering
+       * the operations has never been seen before, we create an entry in the database
+       * with it.
+       */
+      const logger = opts.logger.child({
+        service: ctx.service.typeName,
+        method: ctx.method.name,
+      });
+
+      return handleError<PlainMessage<PublishPersistedOperationsResponse>>(logger, async () => {
+        const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
```

Comment:
You need to validate write access here. For that check the auth context.

Reply:
I've added a check, please check if that's how it should look. Thanks!